src/python/grpcio_tests/tests_aio/unit/call_test.py

```diff
@@ -625,6 +625,27 @@ async def test_call_rpc_error(self):
             self.assertTrue(call.done())
             self.assertEqual(grpc.StatusCode.UNAVAILABLE, await call.code())
 
+    async def test_call_rpc_error_with_async_generator(self):
+        request = messages_pb2.StreamingOutputCallRequest()
+
+        # We fail at connection, this gets started but never runs
+        async def request_iterator():
+            for _ in range(_NUM_STREAM_RESPONSES):
+                yield request
+
+        async with aio.insecure_channel(UNREACHABLE_TARGET) as channel:
+            stub = test_pb2_grpc.TestServiceStub(channel)
+            call = stub.StreamingInputCall(request_iterator(),
+                                           timeout=_SHORT_TIMEOUT_S)
+            with self.assertRaises(aio.AioRpcError) as exception_context:
+                await call
+            self.assertEqual(grpc.StatusCode.DEADLINE_EXCEEDED,
+                             exception_context.exception.code())
+
+        self.assertTrue(call.done())
+        self.assertEqual(grpc.StatusCode.DEADLINE_EXCEEDED, await call.code())
+        self.assertTrue(call._async_request_poller.done())
```

Comment:
This test as written doesn't actually demonstrate issue #25339  (maybe because the request_iterator is never started?)